### DISCUSSION TITLE
Using Tailwind standardized configs

### DISCUSSION THREAD

**Discussion by natedunn** [tw-in-js/twind#29] [Ideas]
I wanted to start off by saying congrats. It's really neat idea that handles nearly all of my use cases for using Tailwind in JS projects.

The thing that gives me pause is the inability to use [Tailwind's standardized configuration](https://tailwindcss.com/docs/configuration). While I can appreciate the easy-of-use the `setup` function provides, it can create some confusion in how one might get started with _twind_. Specifically since [Tailwind's standardized plugins](https://tailwindcss.com/docs/plugins) do not seem to be supported, which is pretty problematic for anyone who relies on them already.

I guess at the end of the day my assumption is that this project would be more inline with Tailwind standards than it currently is, rather than creating more complications or diversions from Tailwind norms. My dream scenario might be to simply provide a path to the config file through the `setup` function:

```js
setup({
  config: "tailwind.config.js"
})
```

**Comment by sastan** [comment]
Thanks for the input. We see the same problem and are planning to support as much of the config as possible. The theme config is already almost the same and plugins is definitely on the roadmap before a v1 release. It will be difficult to support every config option but we are aiming to support the most common use case with alternatives for the missing pieces.

Loading a configuration file in the way you are proposing it will not work as we run mostly in the browser. The following already works for theme only configurations:

```js
import config from 'tailwind.config.js'

setup(config)
```

It would be a great help if you could give us some insight into which plugins you are using and which other configuration options you would require.

**Comment by sastan** [comment]
I have created an issue #32 to track what needs to be done to support tailwind plugins. Feel free to add your input.

**Comment by lukejacksonn** [comment]
Yes, I agree that it is important that configs are as compatible as possible. Our goal was to make them as "copy/paste" friendly as possible (and for the most part they are) but plugins are an outlier here as we created our own [plugin syntax](https://github.com/tw-in-js/twind/blob/main/docs/plugins.md) to work with the compiler.

That said, we have talked about the possibility of writing essentially a transpiler that takes Tailwind plugins and turns them into Twind plugins. This is what @sastan is working toward with #32.

If you find any other discrepancies then please create an issue for them so that we know what to address first. I imagine a lot of the non-plugin differences will be quick fixes. Recently I have been trying to get a clone of https://github.com/tailwindlabs/tailwindcss.com working with Twind as I think it is a good _real world example_.
